apps/backend/src/modules/project-groups/project-group.repository.ts

```diff
@@ -0,0 +1,82 @@
+import { transaction } from "objection";
+
+import { changeCase } from "~/libs/helpers/helpers.js";
+import { HTTPCode } from "~/libs/modules/http/libs/enums/enums.js";
+import { type Repository } from "~/libs/types/types.js";
+
+import { ExceptionMessage } from "./libs/enums/enums.js";
+import { ProjectGroupError } from "./libs/exceptions/exceptions.js";
+import { ProjectGroupEntity } from "./project-group.entity.js";
+import { type ProjectGroupModel } from "./project-group.model.js";
+
+class ProjectGroupRepository implements Repository {
+	private projectGroupModel: typeof ProjectGroupModel;
+
+	public constructor(projectGroupModel: typeof ProjectGroupModel) {
+		this.projectGroupModel = projectGroupModel;
+	}
+
+	public async create(entity: ProjectGroupEntity): Promise<ProjectGroupEntity> {
+		const { name, permissions, projectId, users } = entity.toNewObject();
+		const key = changeCase(name, "snakeCase");
+
+		const trx = await transaction.start(this.projectGroupModel.knex());
```

Comment:
It looks like we have only one query here, do we need this transaction at all in this case 🤔

Reply:
But if an error occurs, for example if the user id does not exist, trx helps to return the database to the state it was in before the transaction started, because otherwise it would add the record with the error to the database

Reply:
When we have only one query to db if it doesn't pass, it won't change anything in it. Transactions are only needed when we have multiple queries, for example, we create groups and then in another query we update user_groups. In this case these are two different queries, so we need to pass transaction to rollback the first query in case the second one fails. For single queries transactions do nothing. In our case I see we have only one query, don't we?

Reply:
okay,
how would you recommend to resolve the problem when the keys from payload are not present in another tables (like projectId in projects etc.)?

Reply:
Nothing additional is needed if this is done in a single query (single unit). But I've just checked objection documentation and it says that insertGraph makes a separate query, so the transaction is recommended, so let's leave it.